components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/impl/AuthenticatorManagementDAOImpl.java

```diff
@@ -0,0 +1,339 @@
+            String authenticatorConfigName, int tenantId) throws AuthenticatorMgtException {
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false);
+        try {
+            return getUserDefinedLocalAuthenticatorByName(dbConnection, authenticatorConfigName, tenantId);
+        } finally {
+            IdentityDatabaseUtil.closeConnection(dbConnection);
+        }
+    }
+
+    @Override
+    public List<UserDefinedLocalAuthenticatorConfig> getAllUserDefinedLocalAuthenticators(int tenantId)
+            throws AuthenticatorMgtException {
+
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true);
+             NamedPreparedStatement statement = new NamedPreparedStatement(dbConnection,
+                Query.GET_ALL_USER_DEFINED_AUTHENTICATOR_SQL)) {
+            statement.setString(Column.DEFINED_BY, DefinedByType.USER.toString());
+            statement.setInt(Column.TENANT_ID, tenantId);
+
+            List<UserDefinedLocalAuthenticatorConfig> allUserDefinedLocalConfigs = new ArrayList<>();
+            try (ResultSet rs = statement.executeQuery()) {
+                while (rs.next()) {
+                    UserDefinedLocalAuthenticatorConfig config = getLocalAuthenticatorConfigBasedOnType(
+                            rs.getString(Column.AUTHENTICATION_TYPE));
+                    config.setName(rs.getString(Column.NAME));
+                    config.setDisplayName(rs.getString(Column.DISPLAY_NAME));
+                    config.setEnabled(rs.getString(Column.IS_ENABLED).equals(IS_TRUE_VALUE));
+                    config.setDefinedByType(DefinedByType.valueOf(rs.getString(Column.DEFINED_BY)));
+
+                    int authenticatorConfigID = getAuthenticatorIdentifier(dbConnection, config.getName(), tenantId);
+                    try (NamedPreparedStatement statementProp = new NamedPreparedStatement(dbConnection,
+                            Query.GET_AUTHENTICATOR_PROP_SQL)) {
+                        statementProp.setInt(Column.AUTHENTICATOR_ID, authenticatorConfigID);
+                        statementProp.setInt(Column.TENANT_ID, tenantId);
+
+                        try (ResultSet rsProp = statementProp.executeQuery()) {
+                            List<Property> properties = new ArrayList<>();
+                            while (rsProp.next()) {
+                                Property property = new Property();
+                                property.setName(rsProp.getString(Column.PROPERTY_KEY));
+                                property.setValue(rsProp.getString(Column.PROPERTY_VALUE));
+                                property.setConfidential(false);
+                                properties.add(property);
+                            }
+                            config.setProperties(properties.toArray(new Property[0]));
+                        }
+                    }
+
+                    allUserDefinedLocalConfigs.add(config);
```

Comment:
check whether we need new indexes

Reply:
We are utilizing the same tables used for identity provider management (IDP_AUTHENTICATOR and AUTHENTICATOR_PROPERTIES) with a subset of their functionality. Therefore, no new index is required.